### PR TITLE
Feature/sp test

### PR DESCRIPTION
**Pull request by xandrej**
__Tasks:__
- [x] done
- [ ] todo
- [ ] got feedback

__Informations:__

| Q            | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Tests pass?  | yes
| Fixed tickets | https://github.com/massiveart/POOL-ALPIN/issues/1149
| Doc          | none

### CODE REVIEW

File: Tests/Functional/Controller/ProductControllerTest.php [hunk 1]
```diff
@@ -1008,4 +1022,28 @@ public function testAllTypeFilter()
...
     }
+
+    public function testPutSpecialPrice()
+    {
+        $data = array(
+            'id' => array('id' => $this->product1->getId()),
+            'status' => array('id' => $this->productStatus1->getId()),
+            'specialPrices' => array(
+            	0 => array(
+                	'price' => $this->specialPrice1->getPrice(),
```
Comment (notsure) [inline Tests/Functional/Controller/ProductControllerTest.php]: syntax

File: Tests/Functional/Controller/ProductControllerTest.php [hunk 2]
```diff
@@ -1008,4 +1022,28 @@ public function testAllTypeFilter()
...
+            	    'currency' => array("code" => $this->specialPrice1->getCurrency()->getCode())
+            	)
+            )
+        );
+
+        $this->client->request('PUT', '/api/products/'.$this->product1->getId(), $data);
+        $this->assertEquals(200, $this->client->getResponse()->getStatusCode());
+
+        $response = json_decode($this->client->getResponse()->getContent());
+        $this->assertEquals('56', $response->specialPrices[0]->price);
```
Comment (notsure) [inline Tests/Functional/Controller/ProductControllerTest.php]: dont you have to use the getter function?  ```$response->specialPrices[0]->getPrice()```

File: Tests/Functional/Controller/ProductControllerTest.php [hunk 3]
```diff
@@ -1008,4 +1022,28 @@ public function testAllTypeFilter()
...
+        $this->assertEquals('eur', $response->specialPrices[0]->currency->code);
```
Comment (notsure) [inline Tests/Functional/Controller/ProductControllerTest.php]: EUR is upper case isn`t it?

File: Tests/Functional/Controller/ProductControllerTest.php (hunk 2 above)
Comment (xandrej) [inline Tests/Functional/Controller/ProductControllerTest.php]: getter function does not work, it's a response array

File: Tests/Functional/Controller/ProductControllerTest.php (hunk 3 above)
Comment (xandrej) [inline Tests/Functional/Controller/ProductControllerTest.php]: getName() gives uppercase getCode() gives lowercase

File: Tests/Functional/Controller/ProductControllerTest.php [hunk 4]
```diff
@@ -1008,4 +1022,28 @@ public function testAllTypeFilter()
...
+    public function testPutSpecialPrice()
+    {
+        $data = array(
+            'id' => array('id' => $this->product1->getId()),
+            'status' => array('id' => $this->productStatus1->getId()),
+            'specialPrices' => array(
+                0 => array(
+                    'price' => $this->specialPrice1->getPrice(),
+                    'start' => $this->specialPrice1->getStart()->format('Y-m-d h:i:s'),
+                    'end'   => $this->specialPrice1->getEnd()->format('Y-m-d h:i:s'),
```
Comment (notsure) [inline Tests/Functional/Controller/ProductControllerTest.php]: remove additional whitespaces after 'end'

File: Tests/Functional/Controller/ProductControllerTest.php [hunk 5]
```diff
@@ -1008,4 +1022,28 @@ public function testAllTypeFilter()
         $this->assertEquals('ProductNumber-1', $response->_embedded->products[0]->number);
         $this->assertEquals('ProductNumber-1', $response->_embedded->products[1]->number);
     }
+
+    public function testPutSpecialPrice()
+    {
+        $data = array(
+            'id' => array('id' => $this->product1->getId()),
+            'status' => array('id' => $this->productStatus1->getId()),
+            'specialPrices' => array(
```
Comment (notsure) [inline Tests/Functional/Controller/ProductControllerTest.php]: array(
    array(
        'price' ...

**Comment by notsure** [comment]
else ack

**Pull request closed without merging**